### PR TITLE
Make causal and attention masks work with < 32-bit widths

### PR DESCRIPTION
**Pull request by danieldk**
## Description

Since we need to get information about the data type using `finfo`, we also have to remove (non-JIT) TorchScript support.

### Types of change

Bugfix

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by shadeMe — APPROVED** [review]

File: curated_transformers/models/attention.py [hunk 1]
```diff
@@ -38,6 +42,34 @@ def shape(self):
...
+
+    :param input:
+        Attention logits to apply the causal mask to. **Shape:**
+        (batch, heads, query_len, key_len)
+    :returns:
+        Logits with the causal mask applied. **Shape:**
+        (batch, heads, query_len, key_len)
+    """
+
+    _, _, query_len, key_len = input.shape
```
[the remark this answers is not in the document]
Comment (danieldk) [reply curated_transformers/models/attention.py:58]: I (somehow) don't expect creating this to be that impactful. I think we should at some point just do regular profiling and identify the bottlenecks that way.
